test/interpreter/function-rate.spec.ts

```diff
@@ -13,29 +15,211 @@ describe('Function RATE', () => {
     expect(engine.getCellValue(adr('B1'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
   })
 
-  it('should calculate the correct value with correct arguments and defaults', () => {
+  it('should return #NUM if algorithm does not converge', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=RATE(12, -100, 100)', ],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.NUM))
+  })
+
+  it('should return #VALUE if guess param is invalid', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=RATE(12, -100, 400, 0, 0, -1)', ],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.VALUE))
+  })
+
+  it('should set the value type to percent', () => {
     const engine = HyperFormula.buildFromArray([
       ['=RATE(12, -100, 400)', ],
-      ['=RATE(12, -100, 400, 100, 1)', ],
-      ['=RATE(12, -100, 400, 1, 1)', ],
-      ['=RATE(12, -100, 400, 0, 1)', ],
-      ['=RATE(12, -100, 400, -100, 1)', ],
-      ['=RATE(12, -100, 400, 100, 1, -1)', ],
```

Comment:
The assertion for the formula presented above is missing from the new tests. I know that you added another assertion (`=RATE(12, -100, 400, 0, 0, -1)`), so disregard this comment if you think it does not have a point.

Reply:
Line 28 of this file

Reply:
Yeah, they are equivalent. They check if an error will be thrown for the `guess` value of `-1`.